leakcanary-analyzer/src/main/java/com/squareup/leakcanary/HeapAnalyzer.java

```diff
@@ -90,6 +95,28 @@ public AnalysisResult checkForLeak(File heapDumpFile, String referenceKey) {
     }
   }
 
+  void dedupGcRoots(Snapshot snapshot) {
+    // THashMap has smaller footprint than HashMap
+    final THashMap<String, RootObj> dedupMap = new THashMap<>();
+
+    final List<RootObj> gcRoots = (ArrayList) snapshot.getGCRoots();
+    for (RootObj root : gcRoots) {
+      String id = root.toString();
```

Comment:
updated the variable name and to manual key generation.  inclined to stick with String.format and optimize later if necessary.